### PR TITLE
Feature Request #106 -  Add Groups - hacktoberfest

### PR DESCRIPTION
**Pull request by jovynode**
@eduayme , all route calls are ajax/stateless for better user experience.  I love the concept of your app. I look forward to contributing more.

### CODE REVIEW

**Comment by jovynode** [comment]
PR #181 sent !

File: resources/lang/en/actions.php [hunk 1]
```diff
@@ -27,4 +27,5 @@
     'filter_by_village' => 'Filter village',
     'filter_by_groups'  => 'Filter groups',
     'filter_by_types'   => 'Filter types',
+    'add_group'         => 'Add Group',
```
Comment (eduayme) [inline resources/lang/en/actions.php:30]: Groups should go undercase. 'add_group' => 'Add group>

File: resources/lang/ca/group.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    'status_active'                 => 'Active',
+    'status_closed'                 => 'Closed',
+    'status_active_value'           => '0',
+    'status_closed_value'           => '1',
+    'lengthMenu'                    => 'Show _MENU_ groups',
+    'group_add_success'             => 'Group was successfully added !',
+    'group_edit_success'            => 'Group was successfully edited !',
+    'group_delete_success'          => 'Group was successfully deleted !',
+    'group_not_found'               => 'Group not found.',
+    'unable_to_delete'               => 'Unable to delete.',
```
Comment (eduayme) [inline resources/lang/ca/group.php:20]: This is not catalan. You should only add the English version or the ones that you know for sure are correct.

File: resources/lang/en/messages.php [hunk 3]
```diff
@@ -23,4 +23,5 @@
     'num_unique'      => 'This number must be unique in the data table.',
     'closed_search'   => 'This investigation is currently closed.',
     'closed_practice' => 'This practice is currently closed.',
+    'no_groups'       => 'No groups have been registered',
```
Comment (eduayme) [inline resources/lang/en/messages.php:26]: A dot in the end is missing. 

'no_groups'       => 'No groups have been registered.',

File: resources/views/searches/resources/groups/add_group.blade.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<!-- Add group button - OPEN -->
+
```
Comment (eduayme) [inline resources/views/searches/resources/groups/add_group.blade.php:2]: The button missing

File: resources/views/searches/resources/groups/edit_group.blade.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
+
+<!-- Add group modal - OPEN -->
```
Comment (eduayme) [inline resources/views/searches/resources/groups/edit_group.blade.php:2]: Should be

File: resources/views/searches/resources/groups/edit_group.blade.php [hunk 6]
```diff
@@ -0,0 +1,94 @@
+
```
Comment (eduayme) [inline resources/views/searches/resources/groups/edit_group.blade.php:1]: Remove empty space

File: resources/views/searches/resources/groups/edit_group.blade.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                </div>
+                <!-- Modal footer - CLOSE -->            
+            </form>
+            <!-- Form register - CLOSE -->
+        </div>
+        <!-- Modal content - CLOSE -->     
+    </div>
+    <!-- Modal dialog - CLOSE -->   
+</div>
+<!-- Add group modal - CLOSE -->
```
Comment (eduayme) [inline resources/views/searches/resources/groups/edit_group.blade.php:94]: Should be

File: resources/views/searches/resources/groups/index.blade.php [hunk 8]
```diff
@@ -1,5 +1,186 @@
-<!-- Content - OPEN -->
-<div class="container margin-top padding-bottom">
-    <img src="/img/mockup_groups.png" style="width: 100%" />
-</div>
-<!-- Content - CLOSE -->
+{{ Form::hidden('group_count', $search->groups->count()) }}
+<!-- If NO nogroup - OPEN -->
```
Comment (eduayme) [inline resources/views/searches/resources/groups/index.blade.php:2]: Should be

File: resources/views/searches/resources/groups/index.blade.php [hunk 9]
```diff
@@ -1,5 +1,186 @@
...
+
+            <a href="#addGroupModal" class="btn btn-primary" role="button" data-toggle="modal"
+                <?php if (Auth::user()->profile == 'guest') { ?> style="display:none" <?php } ?> >
+                {{ __('actions.add') . ' ' . __('group.group') }}
+            </a>            
+        </div>
+    </div>
+<!-- If NO group - CLOSE -->
+
+<!-- If exists searches - OPEN -->
```
Comment (eduayme) [inline resources/views/searches/resources/groups/index.blade.php:20]: Should be

File: resources/views/searches/resources/groups/index.blade.php [hunk 10]
```diff
@@ -1,5 +1,186 @@
...
+                    <tbody>
+                    </tbody>        
+                </table>
+            </div>    
+        </div>    
+    </div>
+    <!-- Content - CLOSE -->
+    @include('searches.resources.groups.add_group',['search_id' => $search->id])
+    @include('searches.resources.groups.edit_group')
+    @include('searches.resources.groups.delete_group')
```
Comment (eduayme) [inline resources/views/searches/resources/groups/index.blade.php:76]: This should go inside the table

File: resources/views/searches/resources/groups/index.blade.php [hunk 11]
```diff
@@ -1,5 +1,186 @@
...
+
+
```
Comment (eduayme) [inline resources/views/searches/resources/groups/index.blade.php:78]: Missing

**Review by eduayme — CHANGES_REQUESTED** [review]

File: resources/lang/en/messages.php (hunk 3 above)
Comment (jovynode) [reply resources/lang/en/messages.php:26]: ok

File: resources/views/searches/resources/groups/add_group.blade.php (hunk 4 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/add_group.blade.php:2]: I removed the comment. The addGroupModal now only contains the Modal screen.  It is called by 2 buttons (lines 12 and 40 of groups/index.blade.php.

File: resources/views/searches/resources/groups/edit_group.blade.php (hunk 5 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/edit_group.blade.php:2]: ok

File: resources/views/searches/resources/groups/edit_group.blade.php (hunk 7 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/edit_group.blade.php:94]: ok

File: resources/views/searches/resources/groups/index.blade.php (hunk 10 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/index.blade.php]: I put edit and delete at tbody. add_group should be outside the table as it is used by 2 Add Buttons (1st is when there's no group yet, 2nd when there's group already).

File: resources/views/searches/resources/groups/index.blade.php (hunk 11 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/index.blade.php]: i assume you're referring to whitespace, i removed it already

File: resources/views/searches/resources/groups/index.blade.php (hunk 9 above)
Comment (jovynode) [reply resources/views/searches/resources/groups/index.blade.php]: ok

**Review by eduayme — APPROVED** [review]

**Pull request merged**